src/coreclr/vm/assemblyspec.cpp

```diff
@@ -271,6 +271,15 @@ HRESULT AssemblySpec::InitializeSpec(StackingAllocator* alloc, ASSEMBLYNAMEREF*
         WCHAR* pString;
         int    iString;
         ((STRINGREF) (*pName)->GetSimpleName())->RefInterpretGetStringValuesDangerousForGC(&pString, &iString);
+
+        for (int i = 0; i < iString; i++)
```

Comment:
This can be just:
```
if (wcslen(pString) != iString)
    ThrowHR(FUSION_E_INVALID_NAME);
```

Reply:
Note that I throw regardless if we are going to parse or not. Once the "Init" API goes managed we can remove `BOOL fParse /*=TRUE*/`, since it will be always `false`

I was hesitant about making the check only in a case if we parse or in a case where we accept the input as well. I am still not 100% decided, but I think we should be consistent.

Reply:
I think that the `!fParse` path has more places where the embedded zeros can cause odd behavior.

For this PR - that just deals with parsing, I think that it would be fine to throw only when fParse is true, and leave the rest for the later.

Reply:
Right. It would mean the check will disappear once "Init" entry point becomes completely managed. 
Throwing on \0 in managed parser is trivial since we examine each char anyways.

Reply:
Also need to do a similar check in mono. 
Mono does `string` -> `char*` conversion a lot earlier than CoreClr - in the `AssemblyName` constructor, so the check will need to be done there (for now). 
That code will be gone when we switch as there will not be a mono-specific constructor.